src/Loops.purs

```diff
@@ -86,64 +125,85 @@ type Event = { sample :: Sample, time :: Millis }
 -- | A `Track` is an infinite list of samples.
 -- |
 -- | A `Track` can be created by using the `loop` function, or played
--- | using the `track` function.
+-- | using the `play` function.
 newtype Track = Track (Lazy.List Event)
 
--- | Beats per minute.
--- |
--- | Use `Data.Newtype.wrap` to create a value:
--- |
--- | ```purescript
--- | wrap 120 :: BPM
--- | ```
-newtype BPM = BPM Int
+derive instance newtypeTrack :: Newtype Track _
+
+-- | A generalization of swing.
+-- | To preserve order, requires a monotonic increasing function of numbers.
+-- | And unless the function intersects the 45 degree line infinitely often,
+-- | this will change the speed of the input Track.
+unsafeWarp :: (Number -> Number) -> Track -> Track
```

Comment:
I was thinking that instead of using `Number -> Number` here, we could use `newtype Warp = Warp (Number -> Number)` and provide some safe constructors which always give monotonic increasing functions. We can also provide an `unsafeWarp :: (Number -> Number) -> Warp` function for advanced use cases.